build.sbt

```diff
@@ -674,62 +698,81 @@ lazy val `scio-core` = project
 lazy val `scio-test` = project
   .in(file("scio-test"))
   .dependsOn(
-    `scio-core` % "compile",
-    `scio-avro` % "compile->test"
+    `scio-test-core`,
+    `scio-test-google-cloud-platform`,
+    `scio-test-parquet`
   )
   .settings(commonSettings)
-  .settings(jUnitSettings)
+  .settings(
+    description := "Scio helpers for ScalaTest",
+    // disable mima due to module split
+    mimaPreviousArtifacts := Set.empty
+  )
+
+lazy val `scio-test-core` = project
+  .in(file("scio-test/core"))
+  .dependsOn(`scio-core`)
+  .settings(commonSettings)
   .settings(macroSettings)
-  .settings(protobufSettings)
   .settings(
     description := "Scio helpers for ScalaTest",
-    undeclaredCompileDependenciesFilter := NothingFilter,
-    unusedCompileDependenciesFilter -= Seq(
-      // added by plugin
-      moduleFilter("com.google.protobuf", "protobuf-java"),
-      // umbrella module
-      moduleFilter("org.scalatest", "scalatest"),
-      // implicit usage not caught
-      moduleFilter("com.spotify", "magnolify-guava"),
-      // junit is required by beam but marked as provided
-      moduleFilter("junit", "junit")
-    ).reduce(_ | _),
+    undeclaredCompileDependenciesFilter -= moduleFilter("org.scalatest"),
+    unusedCompileDependenciesFilter -= moduleFilter("org.scalatest", "scalatest"),
     libraryDependencies ++= Seq(
-      "com.google.api.grpc" % "proto-google-cloud-bigtable-v2" % googleCloudProtoBigTableVersion,
-      "com.google.http-client" % "google-http-client" % googleHttpClientVersion,
+      "com.google.http-client" % "google-http-client" % googleHttpClientVersion, // TODO should we have this here ?
```

Comment:
Possible to move to `google-cloud-platform` test module?

Reply:
no. This is required by the `Pretty` module. Instead of using `toString` in the scio SCollection matchers, we rely on this with custom formatting for
- avro
- gson

Ideally we should use smth like the cat's `Show` typeclass so any etension can give its formatting. Kept as is to avoid breaking the test API.